### PR TITLE
Add Mock Test: Uptime 

### PR DESCRIPTION
**Pull request by Sanket-0510**
refers issue #213 
Add Mocked Tests: Uptime
add tests for uptime.lists_check , uptime.create_check

In API resources the method name is given by the name checks_list(), but the actual method name is list_checks(), thats a point to be noted and hence I have used the method list_check() while writing the tests, correct me if I am wrong.

### CODE REVIEW

**Comment by Sanket-0510** [comment]
@danaelhe am I on the right path.
Right now have added test's for two methods, need a green flag from your side to add the remaining tests for the uptime, thank you.

**Comment by danaelhe** [comment]
Looks great!! Apologies, I should've included in the instructions but we lint our tests with the python package black. Currently, the "lint" test is failing for this PR but it's a super easy fix.  If you run: poetry run black tests while in the pydo directory, it will automatically reformat the files and the "lint" test will pass. So just run that command and push up the changes and the lint tests will pass 👍

**Review by danaelhe — CHANGES_REQUESTED** [review]
Looks SO GOOD! Just need to push up the commit that fixes the lint tests 👍

File: tests/mocked/test_uptime.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
+import responses
+
+from pydo import Client
+
+
+@responses.activate
+def test_uptime_check_lists_get(mock_client: Client, mock_client_url):
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:7]: if you run `make lint-tests`, you'll see a bunch of errors asking:
`tests/mocked/test_uptime.py:42:0: C0116: Missing function or method docstring (missing-function-docstring)`
This is super easy fix, you'll just have to add "docstrings" under the function, as such:
```suggestion
def test_check_state_get(mock_client: Client, mock_client_url):
    """Mock Uptime Check State"""
```

File: tests/mocked/test_uptime.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        f"{mock_client_url}/v2/uptime/checks",
+        json=expected,
+        status=200
+    )
+    get_res = mock_client.uptime.list_checks()
+    assert get_res == expected
+
+
+@responses.activate
+def test_check_state_get(mock_client: Client, mock_client_url):
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:48]: if you run make lint-tests, you'll see a bunch of errors asking:
tests/mocked/test_uptime.py:42:0: C0116: Missing function or method docstring (missing-function-docstring)
This is super easy fix, you'll just have to add "docstrings" under the function, as such:

```suggestion
def test_check_state_get(mock_client: Client, mock_client_url):
    """Mock Uptime Check State"""
```

File: tests/mocked/test_uptime.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        status=200
+    )
+
+    get_res = mock_client.uptime.get_check_state(CHECK_ID)
+
+    assert get_res == expected
+
+
+@responses.activate
+def test_update_check_put(mock_client: Client, mock_client_url):
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:85]: add docstring here as well:
```suggestion
def test_update_check_put(mock_client: Client, mock_client_url):
    """Mock Uptime Check Update"""
```

File: tests/mocked/test_uptime.py [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+        ],
+        "enabled": True
+    }
+
+    put_res = mock_client.uptime.update_check(CHECK_ID, body=req)
+    assert put_res == expected
+
+
+@responses.activate
+def test_list_all_alerts_get(mock_client: Client, mock_client_url):
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:122]: Add docstring here as well:
```suggestion
def test_list_all_alerts_get(mock_client: Client, mock_client_url):
    """Mock Uptime Check List All Alerts"""
```

File: tests/mocked/test_uptime.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        status=200
+    )
+
+    get_res = mock_client.uptime.list_alerts(CHECK_ID)
+
+    assert get_res == expected
+
+
+@responses.activate
+def test_update_alert_put(mock_client: Client, mock_client_url):
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:171]:
```suggestion
def test_update_alert_put(mock_client: Client, mock_client_url):
    """Mock Uptime Update Alert"""
```

File: tests/mocked/test_uptime.py [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+                "started_at": "2022-03-17T18:04:55Z",
+                "ended_at": "2022-03-17T18:06:55Z",
+                "duration_seconds": 120
+            }
+        }
+    }
+    CHECK_ID = "4de7ac8b-495b-4884-9a69-1050c6793cd6"
+    responses.add(
+        responses.GET,
+        f"{mock_client_url}/v2/uptime/checks/{CHECK_ID}/state",
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:74]:
```suggestion
        f"{mock_client_url}/v2/uptime/checks/{check_id}/state",
```

File: tests/mocked/test_uptime.py [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+        json=expected,
+        status=200
+    )
+
+    get_res = mock_client.uptime.get_check_state(CHECK_ID)
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:79]:
```suggestion
    get_res = mock_client.uptime.get_check_state(check_id)
```

File: tests/mocked/test_uptime.py [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+            "type": "https",
+            "target": "https://www.landingpage.com",
+            "regions": [
+                "us_east",
+                "eu_west"
+            ],
+            "enabled": True
+        }
+    }
+    CHECK_ID = "4de7ac8b-495b-4884-9a69-1050c6793cd6"
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:99]:
```suggestion
    check_id = "4de7ac8b-495b-4884-9a69-1050c6793cd6"
```

File: tests/mocked/test_uptime.py [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+        "type": "https",
+        "target": "https://www.landingpage.com",
+        "regions": [
+            "us_east",
+            "eu_west"
+        ],
+        "enabled": True
+    }
+
+    put_res = mock_client.uptime.update_check(CHECK_ID, body=req)
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:117]:
```suggestion
    put_res = mock_client.uptime.update_check(check_id, body=req)
```

File: tests/mocked/test_uptime.py [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+                    "first": "https://api.digitalocean.com/v2/account/keys?page=1",
+                    "prev": "https://api.digitalocean.com/v2/account/keys?page=2"
+                }
+            }
+        },
+        "meta": {
+            "total": 1
+        }
+    }
+    CHECK_ID = "4de7ac8b-495b-4884-9a69-1050c6793cd6"
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:157]:
```suggestion
    check_id = "4de7ac8b-495b-4884-9a69-1050c6793cd6"
```

File: tests/mocked/test_uptime.py [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+    responses.add(
+        responses.GET,
+        f"{mock_client_url}/v2/uptime/checks/{CHECK_ID}/alerts",
+        json=expected,
+        status=200
+    )
+
+    get_res = mock_client.uptime.list_alerts(CHECK_ID)
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:165]:
```suggestion
    get_res = mock_client.uptime.list_alerts(check_id)
```

File: tests/mocked/test_uptime.py [hunk 12]
```diff
@@ -0,0 +1,224 @@
...
+                    {
+                        "channel": "Production Alerts",
+                        "url": "https://hooks.slack.com/services/T1234567/AAAAAAAA/ZZZZZZ"
+                    }
+                ]
+            },
+            "period": "2m"
+        }
+    }
+    CHECK_ID = "4de7ac8b-495b-4884-9a69-1050c6793cd6"
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:193]:
```suggestion
    check_id = "4de7ac8b-495b-4884-9a69-1050c6793cd6"
```

File: tests/mocked/test_uptime.py [hunk 13]
```diff
@@ -0,0 +1,224 @@
...
+    ALERT_ID= "17f0f0ae-b7e5-4ef6-86e3-aa569db58284"
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:194]:
```suggestion
    alert_id= "17f0f0ae-b7e5-4ef6-86e3-aa569db58284"
```

File: tests/mocked/test_uptime.py [hunk 14]
```diff
@@ -0,0 +1,224 @@
...
+    responses.add(
+        responses.PUT,
+        f"{mock_client_url}/v2/uptime/checks/{CHECK_ID}/alerts/{ALERT_ID}",
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:197]:
```suggestion
        f"{mock_client_url}/v2/uptime/checks/{check_id}/alerts/{ALERT_ID}",
```

File: tests/mocked/test_uptime.py [hunk 15]
```diff
@@ -0,0 +1,224 @@
...
+                    "channel": "Production Alerts",
+                    "url": "https://hooks.slack.com/services/T1234567/AAAAAAAA/ZZZZZZ"
+                }
+            ]
+        },
+        "period": "2m"
+    }
+
+    put_res = mock_client.uptime.update_alert(
+        CHECK_ID, ALERT_ID, body=req)
```
Comment (danaelhe) [inline tests/mocked/test_uptime.py:221]:
```suggestion
        check_id, alert_id, body=req)
```

**Comment by Sanket-0510** [comment]
Fixed the lint errors but when I run the command `poetry run black tests` and formatted the code. Running `make lint-tests` I am getting an error or a warning maybe,  am not clear: "line too long" like please look into this and suggest the further changes if any required.

**Review by danaelhe — APPROVED** [review]
AWESOME job!! This is good to merge and ship 🚀 

Thank you so much for your contribution!

**Pull request merged**